src/sentry_sdk.cpp

```diff
@@ -0,0 +1,139 @@
+#include "sentry_sdk.h"
+
+#include "sentry/contexts.h"
+#include "sentry/disabled_sdk.h"
+#include "sentry/uuid.h"
+#include "sentry_options.h"
+
+#include <godot_cpp/classes/os.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+#ifdef NATIVE_SDK
+#include "sentry/native/native_sdk.h"
+#endif
+
+using namespace godot;
+using namespace sentry;
+
+SentrySDK *SentrySDK::singleton = nullptr;
+
+VARIANT_ENUM_CAST(Level);
+
+void SentrySDK::capture_message(const String &p_message, Level p_level, const String &p_logger) {
+	internal_sdk->capture_message(p_message, p_level, p_logger);
+}
+
+void SentrySDK::add_breadcrumb(const String &p_message, const String &p_category, Level p_level,
+		const String &p_type, const Dictionary &p_data) {
+	internal_sdk->add_breadcrumb(p_message, p_category, p_level, p_type, p_data);
+}
+
+String SentrySDK::get_last_event_id() const {
+	return internal_sdk->get_last_event_id();
+}
+
+void SentrySDK::set_tag(const String &p_key, const String &p_value) {
+	ERR_FAIL_COND_MSG(p_key.is_empty(), "Sentry: Can't set tag with an empty key.");
```

Comment:
Is this a user failing failure? how does it manifest?

We have a rule to not have the SDK error/crash the users' app. No one likes their crash reporting service being the reason why their customers had issues. So what we try to do is rely heavily on the `option.debug` flag so that if that's set to true, anything going on with the SDK that could be useful for debugging the SDK is made available to the end user (e.g: console log).

So in this case, we'd log to console (if debug.true) and no-op this call (just return)

Reply:
This is a no-op macro, so it aligns with what you described. Read it like this: If the condition is true, return from the function while also reporting it as an error via the godot logger with a custom message.
This macro would expand to something like this:
```cpp
if (__builtin_expect(!!(p_key.is_empty()), 0)) {
   godot::_err_print_error(__FUNTION__, ..., "Sentry: Can't set tag with an empty key.")
   return;
} else ...
```
There are several such macros: FAIL in this case means return from the function while registering it as an error. Godot has a policy to crash only on unrecoverable issues like some signals. There are also crash macros like  `CRASH_COND(predicate)` - so you'll know when you see such.